base/regex.jl

```diff
@@ -285,6 +285,11 @@ iterate(m::RegexMatch, args...) = iterate(m.captures, args...)
 length(m::RegexMatch) = length(m.captures)
 eltype(m::RegexMatch) = eltype(m.captures)
 
+NamedTuple(m::RegexMatch) = NamedTuple{Symbol.(Tuple(keys(m)))}(values(m))
+Dict(m::RegexMatch) = Dict(pairs(m))
+Dict{String}(m::RegexMatch) = Dict(string.(keys(m)) .=> values(m))
```

Comment:
```suggestion
```
This is not how dictionary constructors are written, with only one type parameter (but this case somewhat already falls out from the previous definition that just calls pairs)